### PR TITLE
SDI-818: Restart failed plugins

### LINKED ISSUE #497: A failed or panicked available plugin can leave an apPool with not enough resources.
When a plugin panics while it is in use by a task, it emits a `DeadAvailablePlugin` event.  This event should be handled by runner, and it should apply the same eligibility rules to the affected `apPool`. I.e. start a new available plugin, and insert it in the dead plugin's place.

### PR DESCRIPTION
**Pull request by candysmurf**
resolves #497.

### CODE REVIEW

**Comment by tiffanyfj** [comment]
You need to rebase.

File: control/runner.go [hunk 1]
```diff
@@ -247,9 +252,39 @@ func (r *runner) HandleGomitEvent(e gomit.Event) {
...
 			return
 		}
+
 		if pool != nil {
 			pool.Kill(v.Id, "plugin dead")
 		}
+
+		if pool.Eligible() {
+			if pool.RestartCount() < MaxPluginRestartCount {
+				e := r.restartPlugin(v.Key)
```
Comment (jcooklin) [inline control/runner.go]: Let's add a log statement (warn) that we restarted plugin (name:version) x times.

**Comment by jcooklin** [comment]
LGTM, after adding warn statement when restarting and rebasing.

File: control/runner.go (hunk 1 above)
Comment (candysmurf) [inline control/runner.go]: Add the warn log. please check. thanks

**Pull request merged**